_posts/2020-04-18-schedule-your-functions.md

```diff
@@ -0,0 +1,130 @@
+* [arkade](https://github.com/alexellis/arkade) - one line installation of applications with which we will install OpenFaaS
+* [faas-cli](https://docs.openfaas.com/cli/install/) - the CLI which communicates with the OpenFaaS gateway
+* [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) - the CLI which communicates with your Kubernetes cluster
+
+## Prepare environment
+
+Before showing you how to run our functions on a schedule, first we need to set up the environment with the tools mentioned above.
+
+### Create cluster
+
+First thing we need is running Kubernetes cluster:
+
+```bash
+kind create cluster
+```
+
+Wait for the installation to finish and verify the namespaces are `Active`:
+
+```bash
+kubectl get ns
+```
+
+### Install OpenFaaS
+
+With `arkade` the installation of OpenFaaS boils down to single line command:
+
+```bash
+arkade install openfaas
+```
+
+Wait for `Active` status on all pods:
+
+```bash
+kubectl get pods -n openfaas --watch
+```
+
+### Set up OpenFaaS client
+
+Get the OpenFaaS Gateway password:
+
+```bash
+PASSWORD=$(kubectl get secret -n openfaas basic-auth -o jsonpath="{.data.basic-auth-password}" | base64 --decode; echo)
+```
+
+Locate the external IP of the kubernetes node:
+
+```bash
+EXTERNAL_IP=$(docker inspect -f "{{ .NetworkSettings.IPAddress }}" $(docker ps -aqf "name=kind-control-plane"))
+```
+
+Authenticate the client:
+
+```bash
+echo -n $PASSWORD | faas-cli login --username admin --password-stdin --gateway=http://$EXTERNAL_IP:31112
+```
+
+## Schedule the functions
+
+After we have the environment up and running, now we can install the cron-connector and schedule our functions to run at specific intervals.
+
+### Deploy the cron-connector
+
+In order to install our `cron-connector` and schedule a function run the following command:
+
+```bash
+curl -s https://raw.githubusercontent.com/openfaas-incubator/cron-connector/master/yaml/kubernetes/connector-dep.yml | kubectl apply -f -
+```
+
+Wait for the connector's status to turn `Running`:
+
+```bash
+kubectl get pods -n openfaas --watch
+```
+
+## Run the function
+
+We will run a function from the store once a minute with a `schedule` annotation following this cron syntax `*/1 * * * *` which is the same as in the [linux cron configuration](https://www.cyberciti.biz/faq/how-do-i-add-jobs-to-cron-under-linux-or-unix-oses/).
+
+The cron-connector recognizes which functions should be ran on a schedule by annotation `topic` with value `cron-function`.
+
+Finally we will deploy `node-info` function and apply the schedule and topic annotations mentioned above:
+
+```bash
+faas-cli store deploy nodeinfo --annotation schedule="*/1 * * * *" --annotation topic=cron-function --gateway=http://$EXTERNAL_IP:31112
+```
+
+If you wait for a few minutes and visit the OpenFaaS UI you will see how the `Invocation count` of the function raises every minute.
```

Comment:
This might also be a good us for the `faas-cli logs`.

What if the function used the github search API to count the number of projects with the tag `openfaas` and then print the stats to stdout.  This could be fun and be more interesting in the logs

Reply:
Yeah I agree it would be more interesting, I left a comment on the function used in the blog. My only concern is the focus of the blog, like would we steal the attention from the `cron-connector` if we write a custom function? I already spend half the post showing how you can deploy OpenFaaS

Reply:
I don't think you need go into details about the function, just say you have "written a function to demonstrate the behavior, it is available [here]" and just create a tiny repo that has that function in it (maybe commit the template folder to ensure it builds in the future)

Reply:
Yeah I like this idea, will do :+1: